### DISCUSSION TITLE
Is it feasible to allow for sidebar navigation on Android?

### DISCUSSION THREAD

**Discussion by jgarplind** [callstack/react-native-bottom-tabs#149] [Q&A] [answered]
I'm fairly new to the space of native navigation, so I don't know which limits are imposed on us.

I have this idea that I want tablets in landscape mode to have their navigation on the right side rather than on the bottom/top.

I noticed that this library exposes a `sidebarAdaptable` prop to hook into this native iPad etc. behavior: https://developer.apple.com/documentation/swiftui/sidebaradaptabletabviewstyle

Now I'm curious, is it feasible to support something similar for Android tablets? Based on my brief understanding, it may _not_ be, because this library defers to whatever the default native behavior is, if there even is such a thing as a "default native behavior"?

**Comment by okwasniewski** [comment]
Hey!

Yes I think it's doable, Jetpack Compose offers something like this: https://developer.android.com/develop/ui/compose/layouts/adaptive/build-adaptive-navigation

Where the bottom bar changes to sidebar and then drawer on bigger devices.

This library is not using jetpack compose due to some compatibility issues with React Native / RNScreens but I think it should be possible to recreate this behaviour using non compose views.

**Reply by jgarplind** [reply]
Thanks! Just knowing that the pattern is within reason is nice to know. Again, I'm new in this space, so Jetpack Compose is not something I'm deeply familiar with, yet.

Is the philosophy of this library to defer to defaults, or is it reasonable to go as far as exposing props letting the consumer decide on the logic for when to render different navigation components, and/or where to place them?

In the link you sent there is a section on overriding the default logic, if I understand it correctly: https://developer.android.com/develop/ui/compose/layouts/adaptive/build-adaptive-navigation#customize_navigation_types, so it doesn't seem to be a completely absurd idea.

**Reply by okwasniewski** [reply, marked as answer]
Since this is available in Jetpack Compose (Native way) this library should also support it. Once this is added we could add a function to determine which type of view to show manually. Thanks for this suggestion

**Reply by jgarplind** [reply]
Thanks for being so receptive! Looking forward to using that functionality in the future :)

**Comment by okwasniewski** [comment]
@jgarplind I've got a POC of this working :) 

https://x.com/o_kwasniewski/status/1858620684706214095

It's a pretty big change and still pretty unstable, might take a while to get it released

**Reply by jgarplind** [reply]
Looking good! 🎉

**Reply by micahlt** [reply]
Out of curiosity, are there any updates on this?

**Reply by okwasniewski** [reply]
Yeah, the approach with Jetpack compose wasn't too stable + it added a lot of MB to bundle size. Recent refactor of Android code unblocks sidebar support with this architecture. For now there is not much movement, but I'm hoping to get back to this feature sometime soon

**Reply by micahlt** [reply]
Got it, thanks for the update!
